src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs

```diff
@@ -865,13 +867,76 @@ internal static string ResolvePackagePath(string searchDirectory, string searchP
             {
                 // If the search does not contain wild cards, and the target path shares the same extension, copy it
                 // e.g. <file src="ie\css\style.css" target="Content\css\ie.css" /> --> Content\css\ie.css
-                return targetPath;
+                packagePath = targetPath;
+                targetPath = null;
             }
             else
             {
                 packagePath = Path.GetFileName(fullPath);
             }
-            return Path.Combine(targetPath ?? String.Empty, packagePath);
+
+            string path = Path.Combine(targetPath ?? string.Empty, packagePath);
```

Comment:
Looking through PackTaskLogic, there are FrameworkAssemblyRefs that are also gonna have this problem. 

https://github.com/NuGet/NuGet.Client/blob/343184861e337a695fffb0cf05be4aee23ae6823/src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTaskLogic.cs#L332. 

AssemblyReferences are not a thing in netcore though, so it's only really, really custom scenarios that'll be broken.

It's safe to fix later